mlflow/store/local_artifact_repo.py

```diff
@@ -1,13 +1,22 @@
 import distutils.dir_util as dir_util
 import shutil
 
+from six.moves import urllib
+
 from mlflow.store.artifact_repo import ArtifactRepository
 from mlflow.utils.file_utils import mkdir, list_all, get_file_info
 from mlflow.utils.validation import path_not_unique, bad_path_message
 
 
 class LocalArtifactRepository(ArtifactRepository):
     """Stores artifacts as files in a local directory."""
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        scheme = urllib.parse.urlparse(self.artifact_uri).scheme
+        if scheme != "":
+            self.artifact_dir = self.artifact_uri[len(scheme + "://"):]
```

Comment:
TODO: Test file: path, noticed a test used this convention and it is unsupported.

A possible solution can depend on the os, where parsed_url.netloc is used for windows and parsed_url.path for linux based OSes